blocks/procedures.js

```diff
@@ -1031,6 +937,10 @@ const procedureCallerGetDefMixin = function() {
           (proc) => proc.getName() === name);
       if (!model) return null;
 
+      const returnTypes = model.getReturnTypes();
+      const hasMatchingReturn = this.hasReturn ? returnTypes : !returnTypes;
+      if (!hasMatchingReturn) return this.createDef_(name, params);
```

Comment:
Mmm the built-in procedure blocks only have a return type of `[]` (empty array) or `null`. The interface's type definition is just a bit more expansive. Since this logic lives in the blocks itself, and not in any of our core logic, I don't think it matters that this is testing explicitly for the types of the built-in procedure blocks, because if the types aren't exactly `[]` or `null` that means an external dev has defined their own procedure blocks, which means this code won't be hit anyway.

But on the other hand, you're right that it doesn't match the type definition, which can be confusing. So from a pure readability standpoint I'm happy to change it. Thoughts?

Reply:
I don't think it makes sense to switch this over right now, because external devs houldn't be hitting this code. But I made the new `hasReturn` property private (`hasReturn_`) so that we can easily switch it out later =)

Reply:
Ah makes sense, and sounds good.